ArcadiaParties-SPA/src/app/content/content.component.ts

```diff
@@ -1,16 +1,47 @@
-import { Component } from '@angular/core';
-import { Store } from '@ngrx/store';
+import { Component, OnDestroy } from '@angular/core';
+import { Store, select } from '@ngrx/store';
+import { ActivatedRoute } from '@angular/router';
+import { merge, Subscription } from 'rxjs';
+import { takeUntil, map, filter, debounceTime } from 'rxjs/operators';
 
 import { changeTitleAction } from '../actions/actions';
-import { State } from '../reducers/reducers';
+import { selectUser } from '../user/selector/selector';
+import { changeDepartmentIdAction } from '../department/actions/actions';
+import { selectCurrentDepartment } from '../department/selector/selector';
+import { DepartmentState } from '../department/reducers/reducer';
 
 @Component({
   selector: 'app-content',
   templateUrl: './content.component.html',
   styleUrls: ['./content.component.scss']
 })
-export class ContentComponent {
-  constructor(private store: Store<State>) {
+export class ContentComponent implements OnDestroy {
+  private mergedObservableSubscription: Subscription;
+
+  currentDepartmentId$ = this.store.pipe(select(selectCurrentDepartment));
+
+  constructor(private store: Store<DepartmentState>, route: ActivatedRoute) {
     store.dispatch(changeTitleAction({ title: 'Arcadia Parties' }));
+
+    const routeParams$ = route.params.pipe(filter(params => params.departmentId));
+
+    const currentUserDepartment$ = this.store
+      .pipe(select(selectUser))
+      .pipe(
+        takeUntil(routeParams$),
+        map(user => user.department.id)
+      );
+
+    const mergedObservable$ = merge(
+      currentUserDepartment$,
+      routeParams$.pipe(map(params => +params.departmentId))
```

Comment:
You can move `map` pipe to `routeParams$` variable definition as it is done for `currentUserDepartmentId$`.

Reply:
Map pipe has been moved